src/ILCompiler.ReadyToRun/src/Compiler/DependencyAnalysis/ReadyToRun/SignatureBuilder.cs

```diff
@@ -269,6 +269,10 @@ public void EmitTypeSignature(TypeDesc typeDesc, SignatureContext context)
                     {
                         EmitElementType(CorElementType.ELEMENT_TYPE_OBJECT);
                     }
+                    else if (typeDesc.IsCanonicalSubtype(CanonicalFormKind.Specific))
```

Comment:
Is this supposed to emit `ELEMENT_TYPE_CANON_ZAPSIG` for e.g. `SomeStruct<__Canon>`, or is this specifically just for `__Canon`? It feels like this should just be the latter, but I'm no expect in CoreCLR. This check with cover both former and latter.

Reply:
Per our offline discussion I have modified this to use IsCanonicalDefinitionType in the latest commit.